sklearn/metrics/tests/test_common.py

```diff
@@ -1733,67 +1733,75 @@ def test_metrics_pos_label_error_str(metric, y_pred_threshold, dtype_y_str):
 
 
 def check_array_api_metric(
-    metric, array_namespace, device, dtype, y_true_np, y_pred_np, sample_weight=None
+    metric, array_namespace, device, dtype_name, y_true_np, y_pred_np, sample_weight
 ):
     xp = _array_api_for_tests(array_namespace, device)
+
+    # always test `sample_weight=None` case
+    if sample_weight is not None:
+        check_array_api_metric(
+            metric,
+            array_namespace,
+            device,
+            dtype_name,
+            y_true_np,
+            y_pred_np,
+            sample_weight=None,
+        )
+        sample_weight = xp.asarray(sample_weight, device=device)
```

Comment:
As I said IRL, I would rather avoid such clever recursive calls that hides the intention of the original code when reading a traceback in the CI logs in case of test failures) and better use a flatter, more direct/explicit style in the caller (even if slightly more redundant).